File: go/builtins/funcNow.go
package builtins

import (
	"time"

	"github.com/jplorg/jpl/go/v2/jpl"
)

var funcNow = enclose(func(runtime jpl.JPLRuntime, signal jpl.JPLRuntimeSignal, next jpl.JPLPiper, input any, args ...any) ([]any, error) {
	return next.Pipe(float64(time.Now().UnixMilli()))
})

File: js/lib/builtins/funcNow.ts
import { type JPLRuntimeSignal } from '@/library';
import type JPLRuntime from '@/runtime';

export default function builtin(
  _runtime: JPLRuntime,
  _signal: JPLRuntimeSignal,
  next: (output: unknown) => Promise<unknown[]> | unknown[],
  _input: unknown,
): Promise<unknown[]> | unknown[] {
  return next(Date.now());
}
